common/mock/marshalizerMock.go

```diff
@@ -22,6 +23,10 @@ func (mm *MarshalizerMock) Marshal(obj interface{}) ([]byte, error) {
 		return nil, ErrMockMarshalizer
 	}
 
+	if mm.MarshalCalled != nil {
```

Comment:
we already have MarshalizerStub in the same package that does that, so maybe reuse that implementation